extensions/exquery/restxq/src/main/java/org/exist/extensions/exquery/restxq/impl/RestXqServiceImpl.java

```diff
@@ -164,9 +171,24 @@ protected Sequence extractRequestBody(final HttpRequest request) throws RestXqSe
                     if (contentType.contains(";")) {
                         contentType = contentType.substring(0, contentType.indexOf(";"));
                     }
-
                     MimeType mimeType = MimeTable.getInstance().getContentType(contentType);
-                    if (mimeType != null && !mimeType.isXMLType()) {
+
+                    if(contentType.equalsIgnoreCase("multipart/related")){
```

Comment:
NPE check please

Reply:
it will be done

Reply:
contentType is checked for null value  [here](https://github.com/eXist-db/exist/pull/1552/files#diff-c29b4b471e6ee7611c4de8e42c6df4c6R169).
I hope we mean the same.